File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/LiveData.swift
//
//  LiveData.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import UIKit

class LiveData: TKExamplesExampleViewController {

    let chart = TKChart()
    var dataPoints = [TKChartDataPoint]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)

        NSTimer.scheduledTimerWithTimeInterval(0.127, target: self, selector: #selector(LiveData.updateChart), userInfo: nil, repeats: true)
    }
    
    func updateChart () -> Void {
        chart.removeAllData()
        let dataPoint = TKChartDataPoint(x: NSDate(), y: Int(arc4random() % 70))
        dataPoints.append(dataPoint)
        if dataPoints.count > 25 {
            dataPoints.removeAtIndex(0)
        }
        
        chart.yAxis = TKChartNumericAxis(minimum: 0, andMaximum: 100)
        let firstPoint = dataPoints[0]
        let lastPoint = dataPoints[dataPoints.count - 1]
        let xAxis = TKChartDateTimeAxis(minimumDate: firstPoint.dataXValue as! NSDate, andMaximumDate: lastPoint.dataXValue as! NSDate)
        xAxis.style.labelStyle.fitMode = TKChartAxisLabelFitMode.None
        xAxis.style.majorTickStyle.maxTickClippingMode = TKChartAxisClippingMode.Visible
        xAxis.majorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Seconds
        chart.xAxis = xAxis
        
        chart.addSeries(TKChartLineSeries(items: dataPoints))
        chart.reloadData()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: TelerikUIExamplesInCSharp/Examples/Chart/LiveData.cs
using System;
using System.Drawing;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("LiveData")]
	public class LiveData : XamarinExampleViewController
	{
		TKChart chart;
		List<TKChartDataPoint> dataPoints;
		TKChartLineSeries lineSeries;

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			this.chart = new TKChart (this.View.Bounds);
			this.chart.AutoresizingMask = UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleWidth;
			this.View.AddSubview (chart);
			this.dataPoints = new List<TKChartDataPoint> ();
			NSTimer.CreateRepeatingScheduledTimer(0.127, UpdateChart);
		}

		public void UpdateChart (NSTimer tr) {
			this.chart.RemoveAllData ();
			Random r = new Random ();
			TKChartDataPoint dataPoint = new TKChartDataPoint (NSDate.Now, new NSNumber (r.Next () % 70));
			this.dataPoints.Add (dataPoint);
			if (this.dataPoints.Count > 25) {
				this.dataPoints.RemoveAt (0);
			}

			this.chart.YAxis = new TKChartNumericAxis (new NSNumber (0), new NSNumber (100));
			TKChartDataPoint firstPoint = this.dataPoints [0];
			TKChartDataPoint lastPoint = this.dataPoints [this.dataPoints.Count - 1];
			TKChartDateTimeAxis xAxis = new TKChartDateTimeAxis (firstPoint.DataXValue, lastPoint.DataXValue);
			xAxis.Style.LabelStyle.FitMode = TKChartAxisLabelFitMode.None;
			xAxis.Style.MajorTickStyle.MaxTickClippingMode = TKChartAxisClippingMode.Visible;
			xAxis.MajorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Seconds;
			this.chart.XAxis = xAxis;

			this.lineSeries = new TKChartLineSeries (this.dataPoints.ToArray ());
			this.chart.AddSeries (this.lineSeries);
			this.chart.ReloadData ();
		}
	}
}

